### PR TITLE
fix(dropdown): calculate the position of the dropdown panel

### PR DESCRIPTION
**Pull request by ju-Skinner**
## JIRA
https://kajabi.atlassian.net/browse/DSS-60

## Description
The initial problem was when a dropdown would be invoked and render outside the viewport. This PR addresses that issue, by determining the available viewport space and renders accordingly.

## Screenshots

|  Before  |  After  |
|--------|--------|
image|image|

Loom: [here](https://www.loom.com/share/55ff74d24f134ac59d51f334cb53d146)

## Testing in `kajabi-products`
1. Turn the bridge on for Sage Lib to KP
2. Navigate to a few locations that use dropdown (a few examples below)
  - Products Index
  - Websites
3. Confirm behavior shown in loom video.

### CODE REVIEW

File: packages/sage-system/lib/dropdown.js [hunk 1]
```diff
@@ -167,8 +167,63 @@ Sage.dropdown = (function () {
...
+    }
+
+    const offset = {
+      top: panelDimensions.top + win.pageYOffset,
+      left: panelDimensions.left + win.pageXOffset,
+      bottom: (panelDimensions.top + win.pageYOffset)
+    };
+
+    const enoughSpaceAbove = viewport.top < ( offset.top + getHeight(panel));
+    const enoughSpaceBelow = viewport.bottom > (offset.bottom + getHeight(panel));
```
Comment (QuintonJason) [inline packages/sage-system/lib/dropdown.js:212]: Regarding the positioning, should `left` and `right` also be considered, or is this something that is out of scope for this work?

Comment (ju-Skinner) [reply packages/sage-system/lib/dropdown.js:212]: My opinion is out of scope...But I am open to hear others as well. 

cc: @monicawheeler

**Review by pixelflips — APPROVED** [review]
Unable to check with the bridge but seems to be working fantastic on the docs site as well. Nice work!

File: packages/sage-system/lib/dropdown.js [hunk 2]
```diff
@@ -167,8 +167,63 @@ Sage.dropdown = (function () {
...
+    const docEl = window.document.documentElement;
+
+    panel.style.top = ''; // resets the style
+
+    // Dimensions
+    const buttonDimensions = button.getBoundingClientRect();
+    const panelDimensions = panel.getBoundingClientRect();
+
+    const panelNewLoc = {
+      top: buttonDimensions.height + panelDimensions.height
```
Comment (monkeypox8) [inline packages/sage-system/lib/dropdown.js:197]: Since we're applying this to all dropdown instances, thoughts on accounting for the bottom `margin` in the calculation here? We'd be able to more closely match the default position.

margin

File: packages/sage-system/lib/dropdown.js (hunk 1 above)
Comment (monkeypox8) [reply packages/sage-system/lib/dropdown.js:212]: I'm in favor of leaving it for a follow-up. Seems useful for views using the Drawer component but not an immediate need for this fix.

**Review by monkeypox8 — APPROVED** [review]
👍 LGTM! Noted one item to consider re: the `top` position calculation

File: packages/sage-system/lib/dropdown.js (hunk 1 above)
Comment (QuintonJason) [reply packages/sage-system/lib/dropdown.js:212]: Then you're good to go

**Review by QuintonJason — APPROVED** [review]
Great addition
